source/components/temp_doeclim_component.cpp

```diff
@@ -0,0 +1,528 @@
+        
+    }
+    
+    // Switched on (To switch off, comment out lines below)
+    C[0] = 1.0 / pow(taucfl, 2.0) + 1.0 / pow(taukls, 2.0) + 2.0 / taucfl / taukls + bsi / taukls / tauksl;
+    C[1] = -1 * bsi / pow(taukls, 2.0) - bsi / taucfl / taukls - bsi / taucfs / taukls - pow(bsi, 2.0) / taukls / tauksl;
+    C[2] = -1 * bsi / pow(tauksl, 2.0) - 1.0 / taucfs / tauksl - 1.0 / taucfl / tauksl -1.0 / taukls / tauksl;
+    C[3] = 1.0 / pow(taucfs, 2.0) + pow(bsi, 2.0) / pow(tauksl, 2.0) + 2.0 * bsi / taucfs / tauksl + bsi / taukls / tauksl;
+    for(int i=0; i<4; i++) {
+        C[i] = C[i] * (pow(double(dt), 2.0) / 12.0);
+    }
+    
+    //------------------------------------------------------------------
+    // Matrices of difference equation system B*T(i+1) = Q(i) + A*T(i)
+    // T = (TL,TO)
+    // (Equation A.27, EK05, or Equations 2.3.24 and 2.3.27, TK07)
+    B[0] = 1.0 + double(dt) / (2.0 * taucfl) + double(dt) / (2.0 * taukls);
+    B[1] = double(-dt) / (2.0 * taukls) * bsi;
+    B[2] = double(-dt) / (2.0 * tauksl);
+    B[3] = 1.0 + double(dt) / (2.0 * taucfs) + double(dt) / (2.0 * tauksl) * bsi + 2.0 * fso * pow((double(dt) / taudif), 0.5);
+    
+    A[0] = 1.0 - double(dt) / (2.0 * taucfl) - double(dt) / (2.0 * taukls);
+    A[1] = double(dt) / (2.0 * taukls) * bsi;
+    A[2] = double(dt) / (2.0 * tauksl);
+    A[3] = 1.0 - double(dt) / (2.0 * taucfs) - double(dt) / (2.0 * tauksl) * bsi + Ker[ns-1] * fso * pow((double(dt) / taudif), 0.5);
+    for (int i=0; i<4; i++) {
+        B[i] = B[i] + C[i];
+        A[i] = A[i] + C[i];
+    }
+    
+    // Calculate the inverse of B
+    invert_1d_2x2_matrix(B, IB);
+}
+
+
+//------------------------------------------------------------------------------
+// documentation is inherited
+void TempDOECLIMComponent::run( const double runToDate ) throw ( h_exception ) {
+    H_LOG( logger, Logger::DEBUG ) << "DOECLIM run " << runToDate << std::endl;
+    
+    // Commented section below is for case of user-specified temperature record.
+    // TempDOECLIM component can't handle that at the moment!
+    
+    //// We track total radiative forcing using internal variable `internal_Ftot`
+    //// Need to do this because if we're subject to a user constraint (being forced
+    //// to match a temperature record), need to track the Ftot that *would* have
+    //// produced the observed temperature record. This way there's a smooth
+    //// transition when we exit the constraint period, after which internal_Ftot
+    //// will rise in parallel with the value reported by ForcingComponent.
+    
+    // If we never had any temperature constraint, `internal_Ftot` will match `Ftot`.
+    //
+    // If the user has supplied temperature data, use that (except if past its end)
+    if( tgav_constrain.size() && runToDate <= tgav_constrain.lastdate() ) {
+    //    H_LOG( logger, Logger::NOTICE ) << "** Using user-supplied temperature" << std::endl;
```

Comment:
Delete this line.  Saving an old version of an error message in a commented line is perhaps the most egregious example of code hoarding I can think of.

Reply:
@bpbond @rplzzz we should probably get this constraint working again in the new component.  for now i'll leave the comments in and we can test this later.